src/transport.js

```diff
@@ -53,11 +54,18 @@ module.exports = function (swarm) {
       // specific for the transport we are using
       const proxyConn = new Connection()
 
-      next(multiaddrs.shift())
+      each(multiaddrs, next)
```

Comment:
@dignifiedquire that should work, I like that we can control the concurrency and not have 20 (figurative and literal 😄 ) dials happen at once.